### PR TITLE
LTD-979 Allow checking status of mail

### PR DESCRIPTION
**Pull request by Pithikos**
Changes:
* Adds `Licence.hmrc_mail_status()` to fetch status on the fly from LITE-HMRC
* Adds endpoints for testing (these endpoints cascade to LITE-HMRC).

### CODE REVIEW

**Review by saruniitr — COMMENTED** [review]
Please also squash commits

File: api/licences/tests/test_service.py [hunk 1]
```diff
@@ -34,6 +38,15 @@ def setUp(self):
             licence=self.licence,
         )
 
+    @override_settings(LITE_HMRC_INTEGRATION_URL="http://localhost:8000", LITE_HMRC_INTEGRATION_ENABLED=True)
+    def test_hmrc_mail_status(self):
+        url_params = urllib.parse.urlencode({"id": self.licence.reference_code})
+        url = f"{settings.LITE_HMRC_INTEGRATION_URL}/mail/licence/?{url_params}"
+
+        with requests_mock.Mocker() as m:
+            m.register_uri("GET", url, json={"status": "reply_sent"})
```
[the remark this answers is not in the document]
Comment (Pithikos) [reply api/licences/tests/test_service.py:47]: Don't we just need to test that an email was sent in the E2E tests? Why would we need the rest of fields?

File: api/licences/libraries/hmrc_integration_operations.py [hunk 2]
```diff
@@ -51,6 +58,21 @@ def send_licence(licence: Licence, action: str):
     logging.info(f"Successfully sent licence '{licence.id}', action '{action}' to HMRC Integration")
 
 
+def get_mail_status(licence: Licence):
+    """Get mail status for given licence"""
+
+    url_params = urllib.parse.urlencode({"id": licence.reference_code})
+    url = f"{settings.LITE_HMRC_INTEGRATION_URL}/mail/licence/?{url_params}"
```
[the remark this answers is not in the document]
Comment (saruniitr) [reply api/licences/libraries/hmrc_integration_operations.py:65]: I should've clarified, yes the endpoint exists but it only has `post`, does the default `get` is enough for this or you need to make any changes?

File: api/licences/tests/test_service.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (saruniitr) [reply api/licences/tests/test_service.py:47]: yes but if possible we also need to check the correct type of mail is sent eg it should be of extract type licence data. If we get the filename it is easy to check this as well.

File: api/licences/models.py [hunk 3]
```diff
@@ -37,6 +42,16 @@ class Licence(TimestampableModel):
 
     objects = LicenceManager()
 
+    def hmrc_mail_status(self):
+        """
+        Fetch mail status from HRMC-integration server
+        """
+        from api.licences.libraries.hmrc_integration_operations import get_mail_status
```
Comment (dgnball) [inline api/licences/models.py:49]: Putting an import here seems odd. Why?

File: api/licences/models.py [hunk 4]
```diff
@@ -37,6 +42,16 @@ class Licence(TimestampableModel):
 
     objects = LicenceManager()
 
+    def hmrc_mail_status(self):
```
Comment (dgnball) [inline api/licences/models.py:45]: Are you sure this belongs in the model? I would generally associate model changes with accessing the database which in this case we aren't doing.

**Review by dgnball — APPROVED** [review]

**Pull request merged**